test/modern_bpf/test_suites/syscall_exit_suite/read_x.cpp

```diff
@@ -0,0 +1,62 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_read
+
+#if defined(__NR_close) && defined(__NR_open)
+
+TEST(SyscallExit, readX)
+{
+	auto evt_test = get_syscall_event_test(__NR_read, EXIT_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	/* Open /dev/urandom for reading */
+	int fd = syscall(__NR_open, "/dev/urandom", O_RDONLY);
+	assert_syscall_state(SYSCALL_SUCCESS, "open", fd, NOT_EQUAL, -1);
+
+	/* Read data from /dev/urandom */
+	const unsigned data_len = 64;
+	char buf[data_len];
+	ssize_t read_bytes = syscall(__NR_read, fd, static_cast<void *>(buf), data_len);
+	assert_syscall_state(SYSCALL_SUCCESS, "read", read_bytes, NOT_EQUAL, -1);
+	int errno_value = -errno;
+	if(fd > 0)
```

Comment:
Actually, `fd` could be zero which is a valid file description number. Also I think we do not need the `errno_value` here based that the `assert_syscall_state` will fail anyway.

Reply:
Good catch! Changed to fd >= 0.

However, `errno` won't be set in this case because the syscall will be successful.. Yet, we need to push the result of `read` to userspace even in the successful case, so rather than relying on `errno` (which may not be cleared before the call to read()..), we push 0 to userspace.

I'm going to add another test case that exercises the fail-case of read. I was planning on making another test that tries to read from a file that's been opened `O_WRONLY`, for instance, to exercise the `errno`/error path.